tests/utils/resource_utils.py

```diff
@@ -27,11 +28,22 @@ def object_from_json_relative_path(relative_path):
     return object_from_json(get_resource_content(relative_path))
 
 
-def get_deserialized_object_from_json(relative_path, return_type):
-    resp = namedtuple("MockResp", ["status", "reason"])(200, "")
-    rest_resp = RESTResponse(resp, get_resource_content(relative_path))
+def get_deserialized_object_from_resource(return_type, resource_path):
+    payload = get_resource_content(resource_path)
+    return deserialize_object(return_type, payload)
 
-    return ApiClient().deserialize(rest_resp, (return_type,), True)
+
+def deserialize_object(model, payload):
+    """Deserializes the passed payload to an instance of the specified model
+    Disregards unknown fields is
+
+    :param model: OpenApi generated model
+    :param payload: json payload to be deserialized
+    :return: Instance of the model
+    """
+    response = namedtuple("MockResp", ["status", "reason"])(200, "")
+    response = RESTResponse(response, payload)
+    return ApiClient(configuration=Configuration(discard_unknown_keys=True)).deserialize(response, (model,), True)
```

Comment:
The `configuration=Configuration(discard_unknown_keys=True)` is a tricky one.

Not using it in the core code of the ApiClient() raises an issue with the [list-audit-trail](https://developers.symphony.com/restapi/reference#list-audit-trail-v1) endpoint: The payload for this endpoint returns multiple attributes that are not described in the [API spec](https://github.com/symphonyoss/symphony-api-spec/blob/master/agent/agent-api-public.yaml#L5144) which means if someone uses this endpoint right now, it will raise an Exception.

Same problem for the [create-app](https://developers.symphony.com/restapi/reference#create-app) endpoint.

Conclusion: We need to add it to the ApiClient in the BDK for the list-audit-trail and create-app endpoint to work.

Reply:
In the java BDK unknown properties are ignored: https://github.com/finos/symphony-bdk-java/blob/70745c703147d1ba08eeaacad18868c18c237091/symphony-bdk-http/symphony-bdk-http-jersey2/src/main/java/com/symphony/bdk/http/jersey2/JSON.java#L23
So we could set `Configuration(discard_unknown_keys=True)` in the `ApiClientFactory` for all clients and open a bug ticket for the endpoints whose behavior is inconsistent with documentation and swagger specs.

Reply:
We definitely want the BDK to be able to support unknown keys, for instance if you are targeting a new version of SBE with new attributes being returned you still want the BDK to work

Reply:
I will create a ticket to add that configuration.